### PR TITLE
- All events have been set up for sending and a class is derived from FirebaseMessagingService is created.

### PR DESCRIPTION
**Pull request by yasirinitialcode**
## Related Issue
- A class to extend Firebase Messaging Service has been created.
- Rest of events have been set up. 
- Splash screen landscape mode was disabled.

ISSUE LINK: https://app.asana.com/0/1203144541549209/1204097456360678

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?
- Tested with Redmi Note 7, added screenshot of Firebase push below.

## Screenshots or Sample Output

Screenshot_20230321-121227_One UI Home

### CODE REVIEW

**Review by ghost — APPROVED** [review]
Thank you! Have you tried sending a push from Firebase?

File: app/src/main/java/com/netmera/netmerafintech/utils/NetmeraAnalyticsUtil.kt [hunk 1]
```diff
@@ -40,10 +64,16 @@ object NetmeraAnalyticsUtil: IAnalyticsUtil {
         }
     }
 
-    override fun paymentTransferEvent(whoToTransfer: String) {
-        val event = PaymentTransferEvent()
-        event.setWho(whoToTransfer)
-        Netmera.sendEvent(event)
+    override fun paymentTransferEvent(whoToTransfer: String?) {
```
Comment (bsoykal) [inline app/src/main/java/com/netmera/netmerafintech/utils/NetmeraAnalyticsUtil.kt]: receiver: String, btw why did you make it nullable?

File: app/src/main/java/com/netmera/netmerafintech/utils/IAnalyticsUtil.kt [hunk 2]
```diff
@@ -8,14 +8,26 @@ import com.netmera.netmerafintech.data.model.Card
...
+    fun changeEvent()
+    fun contactsEvent()
     fun forgotYourPinEvent()
     fun freezeCardEvent(card: Card)
     fun getStartedEvent()
     fun manageEvent()
+    fun nearbyEvent()
     fun paymentDetailEvent(index: Int?)
-    fun paymentTransferEvent(whoToTransfer: String)
+    fun paymentTransferEvent(whoToTransfer: String?)
```
Comment (bsoykal) [inline app/src/main/java/com/netmera/netmerafintech/utils/IAnalyticsUtil.kt]: receiver: String

File: app/src/main/java/com/netmera/netmerafintech/utils/FirebaseAnalyticsUtil.kt [hunk 3]
```diff
@@ -60,9 +107,17 @@ object FirebaseAnalyticsUtil: IAnalyticsUtil {
...
+        whoToTransfer?.let {
+            firebaseAnalytics.logEvent(PAYMENT_TRANSFER) {
+                param(TRANSACTION_ID, it)
+            }
+        }
+    }
+
+    override fun paySomeoneEvent() {
+        firebaseAnalytics.logEvent(PAY_SOMEONE) {
+            param(PAY_SOMEONE, "")
```
Comment (bsoykal) [inline app/src/main/java/com/netmera/netmerafintech/utils/FirebaseAnalyticsUtil.kt]: What is the purpose of adding empty strings?

File: app/src/main/java/com/netmera/netmerafintech/utils/FirebaseAnalyticsUtil.kt [hunk 4]
```diff
@@ -75,6 +130,24 @@ object FirebaseAnalyticsUtil: IAnalyticsUtil {
         }
     }
 
+    override fun requestMoneyEvent() {
+        firebaseAnalytics.logEvent(REQUEST_MONEY) {
+            param(REQUEST_MONEY, "")
```
Comment (bsoykal) [inline app/src/main/java/com/netmera/netmerafintech/utils/FirebaseAnalyticsUtil.kt]: ??

File: app/src/main/java/com/netmera/netmerafintech/utils/FirebaseAnalyticsUtil.kt [hunk 5]
```diff
@@ -75,6 +130,24 @@ object FirebaseAnalyticsUtil: IAnalyticsUtil {
...
+        }
+    }
+
+    override fun seeAllEvent() {
+        firebaseAnalytics.logEvent(SEE_ALL) {
+            param(SEE_ALL, "")
```
Comment (bsoykal) [inline app/src/main/java/com/netmera/netmerafintech/utils/FirebaseAnalyticsUtil.kt]: ??

File: app/src/main/java/com/netmera/netmerafintech/utils/FirebaseAnalyticsUtil.kt [hunk 6]
```diff
@@ -75,6 +130,24 @@ object FirebaseAnalyticsUtil: IAnalyticsUtil {
...
+        }
+    }
+
+    override fun searchForPayeesEvent() {
+        firebaseAnalytics.logEvent(SEARCH_FOR_PAYEES) {
+            param(SEARCH_FOR_PAYEES, "")
```
Comment (bsoykal) [inline app/src/main/java/com/netmera/netmerafintech/utils/FirebaseAnalyticsUtil.kt]: ??

File: app/src/main/java/com/netmera/netmerafintech/utils/FirebaseAnalyticsUtil.kt [hunk 7]
```diff
@@ -75,6 +130,24 @@ object FirebaseAnalyticsUtil: IAnalyticsUtil {
...
+        }
+    }
+
     override fun signInEvent() {
         firebaseAnalytics.logEvent(SIGN_IN) {
             param(SIGN_IN, "")
```
Comment (bsoykal) [inline app/src/main/java/com/netmera/netmerafintech/utils/FirebaseAnalyticsUtil.kt]: ??

File: app/src/main/java/com/netmera/netmerafintech/utils/FirebaseAnalyticsUtil.kt [hunk 8]
```diff
@@ -87,6 +160,18 @@ object FirebaseAnalyticsUtil: IAnalyticsUtil {
         }
     }
 
+    override fun somethingWrongEvent() {
+        firebaseAnalytics.logEvent(SOMETHING_WRONG) {
+            param(SOMETHING_WRONG, "")
```
Comment (bsoykal) [inline app/src/main/java/com/netmera/netmerafintech/utils/FirebaseAnalyticsUtil.kt]: ??

File: app/src/main/java/com/netmera/netmerafintech/utils/FirebaseAnalyticsUtil.kt [hunk 9]
```diff
@@ -87,6 +160,18 @@ object FirebaseAnalyticsUtil: IAnalyticsUtil {
...
+        }
+    }
+
+    override fun supportEvent() {
+        firebaseAnalytics.logEvent(SUPPORT) {
+            param(SUPPORT, "")
```
Comment (bsoykal) [inline app/src/main/java/com/netmera/netmerafintech/utils/FirebaseAnalyticsUtil.kt]: ??

File: app/src/main/java/com/netmera/netmerafintech/utils/FirebaseAnalyticsUtil.kt [hunk 10]
```diff
@@ -14,20 +14,61 @@ object FirebaseAnalyticsUtil: IAnalyticsUtil {
...
+    private const val SEE_ALL = "see_all"
     private const val SIGN_IN = "sign_in"
     private const val SIGN_OUT = "sign_out"
+    private const val SOMETHING_WRONG = "something_wrong"
+    private const val SUPPORT = "support"
     private const val TRANSACTION_ID = "transaction_id"
 
+    override fun addNotesEvent() {
+        firebaseAnalytics.logEvent(ADD_NOTES) {
+            param(ADD_NOTES, "")
```
Comment (bsoykal) [inline app/src/main/java/com/netmera/netmerafintech/utils/FirebaseAnalyticsUtil.kt]: ??

File: app/src/main/java/com/netmera/netmerafintech/utils/FirebaseAnalyticsUtil.kt [hunk 11]
```diff
@@ -14,20 +14,61 @@ object FirebaseAnalyticsUtil: IAnalyticsUtil {
...
+        }
+    }
+
+    override fun appSettingsEvent() {
+        firebaseAnalytics.logEvent(APP_SETTINGS) {
+            param(APP_SETTINGS, "")
```
Comment (bsoykal) [inline app/src/main/java/com/netmera/netmerafintech/utils/FirebaseAnalyticsUtil.kt]: ??

File: app/src/main/java/com/netmera/netmerafintech/utils/FirebaseAnalyticsUtil.kt [hunk 12]
```diff
@@ -14,20 +14,61 @@ object FirebaseAnalyticsUtil: IAnalyticsUtil {
...
+        }
+    }
+
+    override fun cardSettingsEvent() {
+        firebaseAnalytics.logEvent(CARD_SETTINGS) {
+            param(CARD_SETTINGS, "")
```
Comment (bsoykal) [inline app/src/main/java/com/netmera/netmerafintech/utils/FirebaseAnalyticsUtil.kt]: ??

File: app/src/main/java/com/netmera/netmerafintech/utils/FirebaseAnalyticsUtil.kt [hunk 13]
```diff
@@ -52,6 +93,12 @@ object FirebaseAnalyticsUtil: IAnalyticsUtil {
         }
     }
 
+    override fun nearbyEvent() {
+        firebaseAnalytics.logEvent(NEARBY) {
+            param(NEARBY, "")
```
Comment (bsoykal) [inline app/src/main/java/com/netmera/netmerafintech/utils/FirebaseAnalyticsUtil.kt]: ??

File: app/src/main/java/com/netmera/netmerafintech/utils/FirebaseAnalyticsUtil.kt [hunk 14]
```diff
@@ -14,20 +14,61 @@ object FirebaseAnalyticsUtil: IAnalyticsUtil {
...
+
+    override fun contactsEvent() {
+        firebaseAnalytics.logEvent(CONTACTS) {
+            param(CONTACTS, "")
+        }
+    }
+
     override fun forgotYourPinEvent() {
         firebaseAnalytics.logEvent(FORGOT_PIN) {
             param(FORGOT_PIN, "")
```
Comment (bsoykal) [inline app/src/main/java/com/netmera/netmerafintech/utils/FirebaseAnalyticsUtil.kt]: ??

File: app/src/main/java/com/netmera/netmerafintech/utils/FirebaseAnalyticsUtil.kt [hunk 15]
```diff
@@ -14,20 +14,61 @@ object FirebaseAnalyticsUtil: IAnalyticsUtil {
...
+
+    override fun changeEvent() {
+        firebaseAnalytics.logEvent(CHANGE) {
+            param(CHANGE, "")
+        }
+    }
+
+    override fun contactsEvent() {
+        firebaseAnalytics.logEvent(CONTACTS) {
+            param(CONTACTS, "")
```
Comment (bsoykal) [inline app/src/main/java/com/netmera/netmerafintech/utils/FirebaseAnalyticsUtil.kt]: ??

File: app/src/main/java/com/netmera/netmerafintech/utils/FirebaseAnalyticsUtil.kt [hunk 16]
```diff
@@ -14,20 +14,61 @@ object FirebaseAnalyticsUtil: IAnalyticsUtil {
...
+
+    override fun cardSettingsEvent() {
+        firebaseAnalytics.logEvent(CARD_SETTINGS) {
+            param(CARD_SETTINGS, "")
+        }
+    }
+
+    override fun changeEvent() {
+        firebaseAnalytics.logEvent(CHANGE) {
+            param(CHANGE, "")
```
Comment (bsoykal) [inline app/src/main/java/com/netmera/netmerafintech/utils/FirebaseAnalyticsUtil.kt]: ??

File: app/src/main/java/com/netmera/netmerafintech/ImpactFintechMessagingService.kt [hunk 17]
```diff
@@ -0,0 +1,20 @@
+package com.netmera.netmerafintech
+
+import com.google.firebase.messaging.FirebaseMessagingService
+import com.google.firebase.messaging.RemoteMessage
+import com.netmera.Netmera
+
+class ImpactFintechMessagingService : FirebaseMessagingService() {
```
Comment (bsoykal) [inline app/src/main/java/com/netmera/netmerafintech/ImpactFintechMessagingService.kt:7]: You should define this class inside the manifest.

**Review by bsoykal — CHANGES_REQUESTED** [review]
Good, but needs improvements.

File: app/src/main/java/com/netmera/netmerafintech/utils/FirebaseAnalyticsUtil.kt (hunk 3 above)
Comment (yasirinitialcode) [reply app/src/main/java/com/netmera/netmerafintech/utils/FirebaseAnalyticsUtil.kt]: `param(String, Bundle), param(String, Array<Bundle>), param(String, Double), param(String, Long), param(String, String) ` these are the possible key value pairs for param, i thought we need keys but not real values, to send event i set empty string for ones i did not implement a logic such as `paymentTransferEvent(receiver)` .

File: app/src/main/java/com/netmera/netmerafintech/utils/NetmeraAnalyticsUtil.kt (hunk 1 above)
Comment (yasirinitialcode) [reply app/src/main/java/com/netmera/netmerafintech/utils/NetmeraAnalyticsUtil.kt]: Name attribute of Favorites data type defined as optional, just made it mandatory.

File: app/src/main/AndroidManifest.xml [hunk 18]
```diff
@@ -20,7 +20,8 @@
         tools:ignore="LockedOrientationActivity">
```
Comment (bsoykal) [inline app/src/main/AndroidManifest.xml:20]: Why do we have something like this in the manifest?

**Review by bsoykal — APPROVED** [review]
You should remove empty params, the rest is LGTM. Thank you.

File: app/src/main/AndroidManifest.xml (hunk 18 above)
Comment (yasirinitialcode) [reply app/src/main/AndroidManifest.xml:20]: Without it `Expecting android:screenOrientation="unspecified" or "fullSensor" for this activity so the user can use the application in any orientation and provide a great experience on Chrome OS devices` warning comes up (draws line yellow backgrounded).

**Pull request merged**